docs/pow.md

```diff
@@ -0,0 +1,71 @@
+# Proof-of-work in the faucet
+
+The faucet uses POW challenges for anti-dos and dynamically shifts the
+difficulty based on the faucet's balance.
+
+Generally, the less bitcoin in the faucet, the more difficult it is to get out
+via solving the POW challenge.
+
+This was initially implemented via a logarithmic based function but since the
+difficulty parameter already exponentially increases the difficulty, we switched
+to a linear approach which is easier to read and is less likely to blow up.
+
+Output: **$y$**, difficulty parameter. Difficulty is $2^y$. Amounts are in sats.
+
+Parameters:
+- **$x$**: Current balance of the faucet. Positive integer.
+- **$m$**: Minimum difficulty parameter, a good default is about 17-20. Between 0 and **$M$**.
+- **$q$**: Amount in sats emitted per successful faucet claim. Positive integer.
+- **$L$**: Difficulty linear increase coeffcient. A good default is 10-25. Positive integer.
+- **$b$**: Minimum balance. It will be computationally impossible to drop the faucet's balance below
+this value. Defaults to 0. Positive integer.
+
+Constants:
+- **$M$**: Maximum difficulty parameter = 255
```

Comment:
Don't bold these. Otherwise they look like vectors or matrices in standard math notation.


```suggestion
- $x$: Current balance of the faucet. Positive integer.
- $m$: Minimum difficulty parameter, a good default is about 17-20. Between 0 and $M$.
- $q$: Amount in sats emitted per successful faucet claim. Positive integer.
- $L$: Difficulty linear increase coefficient. A good default is 10-25. Positive integer.
- $b$: Minimum balance. It will be computationally impossible to drop the faucet's balance below
this value. Defaults to 0. Positive integer.

Constants:
- $M$: Maximum difficulty parameter = 255
```